functional-programming/src/main.ts

```diff
@@ -0,0 +1,84 @@
+import { Either, fromPromise, ap, right, getOrElse, flatten, left } from "./fp/either";
+import { pipe } from "./fp/utils";
+import { fetchClient, fetchExecutor } from "./fetching";
+import { ClientUser, Demand, ExecutorUser } from "./types";
+import { fold, fromNullable } from "./fp/maybe";
+import { distance } from "./utils";
+import { fromCompare, ordNumber, revert } from "./fp/ord";
+
+type Response<R> = Promise<Either<string, R>>
+
+const getExecutor = (): Response<ExecutorUser> => fromPromise(fetchExecutor());
+const getClients = (): Response<Array<ClientUser>> => fromPromise(fetchClient()
+  .then(value => value.map(({ demands, ...restParams }) => ({
+    demands: fromNullable(demands),
+    ...restParams
+  }))));
+
+export enum SortBy {
+  distance = "distance",
+  reward = "reward",
+}
+
+export const show = (sortBy: SortBy) => (clients: Array<ClientUser>) => (executor: ExecutorUser): Either<string, string> => {
+  const rewardPredicate = (item1: ClientUser, item2: ClientUser) => (
```

Comment:
seems that creation of the new functions should be outside of the function ->
all new pure functions should be in the outside functional scope. - no need to create these functions at each function execution (it will be some sort of side effects, but still it's better as for me)